lib/compress.js

```diff
@@ -176,6 +176,21 @@ merge(Compressor.prototype, {
         }
     };
 
+    // we shouldn't compress (1,func)(something) to
+    // func(something) because that changes the meaning of
+    // the func (becomes lexical instead of global).
+    function check_function_call(parent, orig, val) {
```

Comment:
@kzc @michaelficarra how about `maintain_this_binding()`, as per [reference](http://www.ecma-international.org/ecma-262/5.1/#sec-10.3)?

@kzc I think [this](http://www.ecma-international.org/ecma-262/5.1/#sec-11.1.6) might explain why `(a.b)()` isn't any different from `a.b()`...